src/server/index.ts

```diff
@@ -131,16 +131,80 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
         this.initializePromise = this._initialize();
     }
 
-    private updateSyncStatus(
-        driveId: string,
-        status: SyncStatus | null,
+    private getOperationSource(source: StrandUpdateSource) {
+        return source.type === 'local' ? 'push' : 'pull';
+    }
+
+    private getCombinedSyncUnitStatus(
+        syncUnitStatus: SyncUnitStatusObject
+    ): SyncUnitStatus {
+        if (syncUnitStatus.pull === 'INITIAL_SYNC') return 'INITIAL_SYNC';
+        if (syncUnitStatus.push === 'INITIAL_SYNC') return syncUnitStatus.pull;
+
+        const order = ['ERROR', 'MISSING', 'CONFLICT', 'SYNCING', 'SUCCESS'];
```

Comment:
Can this be made type safe in case the SyncStatus possible values change?
This array should be forced to include all the sync status values.
We could also move this logic to a separate method, outside of this class. Would also make unit tests easier.